internal/controller/etcd/register.go

```diff
@@ -0,0 +1,139 @@
+// NOTE:
+// For all conditions the following is applicable:
+// 1. create and delete events are always reconciled irrespective of whether reconcile annotation is present or auto-reconcile has been enabled.
+// 2. generic events are never reconciled. If there is a need in future to react to generic events then this should be changed.
+// Conditions for reconciliation:
+// Scenario 1: {Auto-Reconcile: false, Reconcile-Annotation-Present: false, Spec-Updated: false/true, Status-Updated: false/true, update-event-reconciled: false}
+// Scenario 2: {Auto-Reconcile: false, Reconcile-Annotation-Present: true, Spec-Updated: false, Status-Updated: false, update-event-reconciled: true}
+// Scenario 3: {Auto-Reconcile: false, Reconcile-Annotation-Present: true, Spec-Updated: false, Status-Updated: true, update-event-reconciled: false}
+// Scenario 4: {Auto-Reconcile: false, Reconcile-Annotation-Present: true, Spec-Updated: true, Status-Updated: false, update-event-reconciled: true}
+// Scenario 5: {Auto-Reconcile: false, Reconcile-Annotation-Present: true, Spec-Updated: true, Status-Updated: true, update-event-reconciled: true}
+// Scenario 6: {Auto-Reconcile: true, Reconcile-Annotation-Present: false, Spec-Updated: false, Status-Updated: false, update-event-reconciled: NA}, This condition cannot happen. In case of a controller restart there will only be a CreateEvent.
+// Scenario 7: {Auto-Reconcile: true, Reconcile-Annotation-Present: false, Spec-Updated: true, Status-Updated: false, update-event-reconciled: true}
+// Scenario 8: {Auto-Reconcile: true, Reconcile-Annotation-Present: false, Spec-Updated: false, Status-Updated: true, update-event-reconciled: false}
+// Scenario 9: {Auto-Reconcile: true, Reconcile-Annotation-Present: false, Spec-Updated: true, Status-Updated: true, update-event-reconciled: true}
+// Scenario 10: {Auto-Reconcile: true, Reconcile-Annotation-Present: false, Spec-Updated: false, Status-Updated: false, update-event-reconciled: false}
+// Scenario 11: {Auto-Reconcile: true, Reconcile-Annotation-Present: true, Spec-Updated: false, Status-Updated: false, update-event-reconciled: true}
+// Scenario 12: {Auto-Reconcile: true, Reconcile-Annotation-Present: true, Spec-Updated: true, Status-Updated: false, update-event-reconciled: true}
+// Scenario 13: {Auto-Reconcile: true, Reconcile-Annotation-Present: true, Spec-Updated: false, Status-Updated: true, update-event-reconciled: false}
+// Scenario 14: {Auto-Reconcile: true, Reconcile-Annotation-Present: true, Spec-Updated: true, Status-Updated: true, update-event-reconciled: true}
+func (r *Reconciler) buildPredicate() predicate.Predicate {
+	// If there is no change to spec and status then no reconciliation would happen. This is also true when auto-reconcile
+	// has been enabled. If an operator wishes to force a reconcile especially when no change (spec/status) has been done to the etcd resource
+	// then the only way is to explicitly add the reconcile annotation to the etcd resource.
```

Comment:
```suggestion
	// then the only way is to explicitly add the reconcile annotation: 'gardener.cloud/operation: reconcile' to the etcd resource.
```

Reply:
This is just a comment within the code, and not a docstring for any publicly exposed API/code, so it isn't necessary to expand on the annotation name every time. Those reading the code here would already know what reconcile annotation means, since that's part of the API. If they didn't know about it, they wouldn't be reading the code.